src/main/java/org/spongepowered/api/entity/Entity.java

```diff
@@ -260,6 +341,48 @@
     boolean transferToWorld(UUID uuid, Vector3d position);
 
     /**
+     * Sets the location of this entity to a new position in a world which does
+     * not have to be loaded (but must at least be enabled) and its rotation.
+     *
+     * <p>If the target world is loaded then this is equivalent to
+     * setting the location via {@link TargetedLocationData}.</p>
+     *
+     * <p>If the target world is unloaded but is enabled according to its
+     * {@link WorldProperties#isEnabled()} then this will first load the world
+     * before transferring the entity to that world.</p>
+     *
+     * <p>If the target world is unloaded and not enabled then the transfer
+     * will fail.</p>
+     *
+     * @param worldName The name of the world to transfer to
+     * @param position The position in the target world
+     * @param rotation The rotation
+     * @return True if the teleport was successful
+     */
+    boolean transferToWorld(String worldName, Vector3d position, Vector3d rotation);
+
+    /**
+     * Sets the location of this entity to a new position in a world which does
+     * not have to be loaded (but must at least be enabled) and its rotation.
+     *
+     * <p>If the target world is loaded then this is equivalent to
+     * setting the location via {@link TargetedLocationData}.</p>
+     *
+     * <p>If the target world is unloaded but is enabled according to its
+     * {@link WorldProperties#isEnabled()} then this will first load the world
+     * before transferring the entity to that world.</p>
+     *
+     * <p>If the target world is unloaded and not enabled then the transfer
+     * will fail.</p>
+     *
+     * @param uuid The UUID of the target world to transfer to
+     * @param position The position in the target world
+     * @param rotation The rotation
+     * @return True if the teleport was successful
+     */
+    boolean transferToWorld(UUID uuid, Vector3d position, Vector3d rotation);
```

Comment:
`UUID` is actually still used in the already existing `transferToWorld` methods.